File: backend/app/schemas/video.py
from pydantic import BaseModel
from typing import Optional, Dict, Any
from datetime import datetime

class VideoBase(BaseModel):
    filename: str
    original_filename: Optional[str] = None
    format: Optional[str] = None

class VideoCreate(VideoBase):
    project_id: str
    file_size: Optional[int] = None
    duration: Optional[float] = None
    width: Optional[int] = None
    height: Optional[int] = None
    fps: Optional[int] = None

class VideoTrimRequest(BaseModel):
    start_time: float
    end_time: float

class VideoEditRequest(BaseModel):
    edit_type: str  # trim, filter, add_transition, change_speed, color_grading
    prompt: Optional[str] = None
    parameters: Optional[Dict[str, Any]] = None

class VideoResponse(VideoBase):
    id: str
    project_id: str
    user_id: str
    file_size: Optional[int] = None
    duration: Optional[float] = None
    width: Optional[int] = None
    height: Optional[int] = None
    fps: Optional[int] = None
    s3_url: Optional[str] = None
    thumbnail_url: Optional[str] = None
    audio_track_count: int = 1
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/video.ts
export interface VideoMedia {
  id: string;
  project_id: string;
  user_id: string;
  filename: string;
  original_filename?: string;
  file_size?: number;
  duration?: number;
  width?: number;
  height?: number;
  fps?: number;
  format?: string;
  s3_url?: string;
  thumbnail_url?: string;
  audio_track_count: number;
  created_at: string;
}

export interface TimelineTrack {
  id: string;
  name: string;
  type: 'video' | 'audio' | 'text';
  muted?: boolean;
  volume?: number; // 0 to 100
}

export interface VideoClip {
  id: string;
  name: string;
  start_time: number;
  end_time: number;
  track: 'video' | 'audio' | 'text';
  track_id?: string;
  color?: string;
  text_content?: string;
  audio_url?: string;
  speed?: number;
  volume?: number;
  muted?: boolean;
}

export interface VideoState {
  currentTime: number;
  duration: number;
  isPlaying: boolean;
  playbackRate: number;
  volume: number;
  isMuted: boolean;
  selectedTrackId: string | null;
  tracks: any[];
  history: any[];
  historyIndex: number;
}

